File: Python/integer_middle.py
###FIRST APPROACH
n = int(input())
a = list(map(int, input().split(" ")))

a.sort()
if len(a)%2 == 0:
    M = len(a)/2
else:
    M = (len(a)+1)/2
# print(M)
print(a[int(M)-1], sum([abs(item-a[int(M)-1]) for item in a]))


###SECOND APPROACH
n = int(input())
a = sorted([int(x) for x in input().split()])

M = a[(n - 1) // 2]
ans = 0
for i in range(n):
    ans += abs(a[i] - M)

print(M, ans)

File: C++/integer_middle.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false); cin.tie(0);
    int n;
    cin >> n;
    int a[n];
    for (int i = 0; i < n; i++)
        cin >> a[i];

    sort(a, a + n);

    int M = a[(n - 1) / 2];
    long long int ans = 0;

    for (int i = 0; i < n; i++)
        ans += abs(a[i] - M);

    cout << M << ' ' << ans << '\n';

    return 0;
}